features/steps/argonaut.py

```diff
@@ -100,6 +100,25 @@ def step_impl(context, resource_type, field_name):
                                       resource_type=resource_type)
 
 
+@then(u'one of the following paths exist {field_string} in {resource}')
+def step_impl(context, field_string, resource):
+
+    fields_to_find = field_string.split(",")
+
+    resources = get_resources(context.response.json(), resource)
+
+    found = False
```

Comment:
I think this initialization step needs to happen inside the loop to get the behavior you want. Otherwise one good resource will shadow a pile of bad ones.

Another way to skin this logic would be something like the following (untested):

```py
def some_field_exists(resource, fields):
    return bool([traverse(resource, field) for field in fields])

def step_impl(context, field_string, resource):
    fields_to_find = field_string.split(",")
    resources = get_resources(context.response.json(), resource)
    valid_resource_ids = set([
        res.id for res in resources
        if some_field_exists(res, fields_to_find)])

    all_resource_ids = set([res.id for res in resources])
    invalid_resource_ids = all_resource_ids - valid_resource_ids
    assert len(invalid_resource_ids) == 0, \
        utils.bad_response_assert( # include the list of invalid resource ids here


```

Reply:
Good catch. I implemented as you advised, spent some time banging my head on the reduce function when it didn't work like I expected. Thanks for the direction!

Error output should look like

Assertion Failed: The resources identified by ids (smart-observation-828-weight, smart-observation-828-oxygensaturation, smart-observation-828-temperature, smart-observation-828-heartrate, smart-observation-828-respiratoryrate, smart-observation-828-bmi, smart-observation-828-systolic, smart-observation-828-diastolic, smart-observation-828-height, smart-observation-828-bp) were absentall of the following fields: valueQuantity.value,component.valueQuantity.value,DataAbsentReason